src/main/capi/prepared-c.cpp

```diff
@@ -14,6 +15,47 @@ using duckdb::QueryResultType;
 using duckdb::timestamp_t;
 using duckdb::Value;
 
+idx_t duckdb_extract_statements(duckdb_connection connection, const char *query,
+                                duckdb_extracted_statements *out_extracted_statements) {
+	if (!connection || !query || !out_extracted_statements) {
+		return 0;
+	}
+	auto wrapper = new ExtractStatementsWrapper();
+	Connection *conn = (Connection *)connection;
+	try {
+		wrapper->statements = conn->ExtractStatements(query);
+	} catch (const duckdb::ParserException &e) {
+		wrapper->error = e.what();
+	}
+
+	*out_extracted_statements = (duckdb_extracted_statements)wrapper;
+	return wrapper->statements.size();
+}
+
+duckdb_state duckdb_prepare_extracted_statement(duckdb_connection connection,
+                                                duckdb_extracted_statements extracted_statements, idx_t index,
+                                                duckdb_prepared_statement *out_prepared_statement) {
+	if (!connection || !out_prepared_statement) {
+		return DuckDBError;
+	}
+	Connection *conn = (Connection *)connection;
+	auto source_wrapper = (ExtractStatementsWrapper *)extracted_statements;
+
+	auto wrapper = new PreparedStatementWrapper();
+	wrapper->statement = conn->Prepare(move(source_wrapper->statements[index]));
```

Comment:
Could we detect if `index` if out of bounds here, and add a test for that?

Reply:
✔️